### PR TITLE
Handle exceptions in acme.NewOrder

### PR DESCRIPTION
**Pull request by sjuranek**
Getting urn:ietf:params:acme:error:rateLimited on a single domain, so have to research that but at least this won't crash the Azure App Service

### CODE REVIEW

**Comment by ffMathy** [comment]
If you get rate limited errors, why do you want to ignore them? Why not instead adjust your renewal rate?

**Comment by sjuranek** [comment]
Any unhandled exception in the background crashes the app. So when I installed this service and deployed it my application would no longer stay running and because it was in the background it was difficult to troubleshoot. I plan on 'forking' the project and have the code run when a certain URL is called (e.g. CheckCert). Then we can verify it is all working and get error details instead of having it run in the background and crash the app.

I dug in and for some reason we can't get the PfxBlob back from azure app certs anymore, so the code is then trying to re-create the SSL cert every time the app starts. This cause us to go over the rate. So this pull request is not a big deal to accept - there are bigger issues that need fixed otherwise this service keep crashing the ASP.NET app that it is installed in - at least with my basic ASP.NET Core 2 app on a new Azure app install.

**Comment by ffMathy** [comment]
But ignoring errors like that in my opinion is bad. Of course it should not crash the app, but we must be able to prevent the error from happening in the first place.

The urn:ietf:params:acme:error:rateLimited error will occur if you try to make too many LetsEncrypt calls. I think the default limit is 5 per week.

Is your renewal time set properly to reflect this? How often do you try to renew?

Also, how did it crash your app? The library runs in a background service. That should never result in an app crash.

Your PR might be fine, but we must understand the root cause of the problem before attempting to wrap something around it.

**Comment by sjuranek** [comment]
Yes, ignoring errors is bad - but logging them and prevent the entire app from crashing is still a bit better.  :) 
I'm not sure why that exception was crashing the app. I'm having a hard time getting the details from Azure. I think the renewal service will check the certificate each time the app starts up. If the cert is null/invalid then it tries to renew. Since the PfxBlob is always returning null then every time the app starts we try to renew, and that is what I think it causing the rate violation.

With Azure persistence is it also meant to use another persistence at the same time? So if I use Azure and SQL Server/Custom then maybe this will all work fine.

**Comment by ffMathy** [comment]
It's not meant to use another persistence service, but using multiple is definitely supported.

I looked at your PR again and it almost makes sense. I'll comment on what I mean in the code to elaborate.

File: src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs [hunk 1]
```diff
@@ -141,7 +141,15 @@ private async Task AcquireNewCertificateForDomains(string[] domains)
 		{
 			_logger.LogInformation("Ordering LetsEncrypt certificate for domains {0}.", new object[] { domains });
 
-			var order = await acme.NewOrder(domains);
+		        IOrderContext order = null;
```
Comment (ffMathy) [inline src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs]: Why allow order to be `null` here?

Instead, extend the try/catch so it reaches further.

**Comment by ffMathy** [comment]
Closing, as another PR #11 has provided an option allowing for how errors should be treated.

**Pull request closed without merging**